source/tutorials/nix-language.md

```diff
@@ -0,0 +1,1602 @@
+- Every *Nix file* (`.nix`) contains a single Nix expression.
+
+:::{note}
+To *evaluate* means to transform an expression according to the language rules until no further simplification is possible.
+:::
+
+All examples in this guide are valid Nix expressions.
+Each one is followed by the expected evaluation result.
+
+The following example is a Nix expression adding two numbers:
+
+```nix
+1 + 2
+```
+
+    3
+
+#### Interactive evaluation
+
+Use [`nix repl`][nix-repl] to evaluate Nix expressions interactively (by typing them on the command line):
+
+```console
+nix repl
+```
+
+    Welcome to Nix 2.5.1. Type :? for help.
+
+```console
+nix-repl> 1 + 2
+```
+
+    3
+
+[nix-repl]: https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-repl.html
+
+#### Evaluating Nix files
+
+Use [`nix-instantiate --eval`][nix-instantiate] to evaluate the expression in a Nix file.
+
+```console
+echo 1 + 2 > file.nix
+
+nix-instantiate --eval file.nix
+```
+
+    3
+
+<details><summary>Detailed explanation</summary>
+
+The first command writes `1 + 2` to a file `file.nix` in the current directory.
+The contents of `file.nix` are now `1 + 2`, which you can check with
+
+```console
+cat file.nix
+```
+
+    1 + 2
+
+The second command runs `nix-instantiate` with the `--eval` option on `file.nix`, which reads the file and evaluates the contained Nix expression.
+The resulting value is printed as output.
+
+`--eval` is required to evaluate the file and do nothing else.
+If `--eval` is omitted, `nix-instantiate` expects the expression in the given file to evaluate to a special data type called *Derivation*, which we will come back to at the end of this guide in [](build-tasks).
+
+If you do not need `file.nix` any more, remove it with
+
+```console
+rm file.nix
+```
+
+</details>
+
+:::{note}
+`nix-instantiate --eval` will evaluate `default.nix` if no file name is specified.
+
+```console
+echo 1 + 2 > default.nix
+
+nix-instantiate --eval
+```
+
+    3
+:::
+
+[nix-instantiate]: https://nixos.org/manual/nix/stable/command-ref/nix-instantiate.html
+
+## Reading the Nix language without fear
+
+You will quickly encounter Nix language expressions that may look very complicated.
+
+As with any programming language, the required amount of Nix language code closely matches the complexity of the problem it is supposed to solve, and reflects how well the problem – and its solution – is understood.
+
+Building software is a complex undertaking, and Nix both exposes and allows managing this complexity with the Nix language.
+
+The purpose of the Nix language is to create *build tasks*: precise descriptions of how contents of existing files are used to derive new files.
+
+:::{important}
+A build task in Nix is called *derivation*.
+:::
+
+The Nix language has only few basic constructs which can be combined arbitrarily:
+
+- primitive data types
+
+  as basic building blocks
+
+- compound data types and functions
+
+  to produce and transform complex data
```

Comment:
I'd say this category should be everything else, maybe call it "Expressions that aren't fully evaluated", including things like `(x: x) 1`, `let x = 10; in x`, etc. Evaluating all of these gives a value that's different from the expression

Reply:
I understand what you mean but I really doubt it would help the cause here. For me that kind of wording crosses the boundary of being so technically correct as to confuse unsuspecting beginners.

Sure, being not technically correct will confuse unsuspecting PLT experts. But you're not the intended audience. I'm already taking some liberty with that entire narrative, comparing Nix to JSON, building complex data and all that. These things are just to support that narrative, to create a mental model that is deliberately simplified to focus on the essentials.

I'd love to make it technically correct and still focused and readable, but I'm out of fantasy right now and would love to see a full suggestion how to get there. Feel free to make a PR against this PR, or propose such a significant change after merging.